src/language-handlebars/utils.js

```diff
@@ -41,70 +23,26 @@ function isVoid(node) {
 }
 
 function isWhitespaceNode(node) {
-  return isNodeOfSomeType(node, ["TextNode"]) && !/\S/.test(node.chars);
-}
-
-function isNodeOfSomeType(node, types) {
-  return node && types.includes(node.type);
-}
-
-function isParentOfSomeType(path, types) {
-  const parentNode = path.parent;
-  return isNodeOfSomeType(parentNode, types);
-}
-
-function isPreviousNodeOfSomeType(path, types) {
-  const previousNode = getPreviousNode(path);
-  return isNodeOfSomeType(previousNode, types);
-}
-
-function isNextNodeOfSomeType(path, types) {
-  const nextNode = getNextNode(path);
-  return isNodeOfSomeType(nextNode, types);
-}
-
-function getSiblingNode(path, offset) {
-  const { node } = path;
-  const parentNode = path.parent ?? {};
-  const children =
-    parentNode.children ?? parentNode.body ?? parentNode.parts ?? [];
-  const index = children.indexOf(node);
-  return index !== -1 && children[index + offset];
-}
-
-function getPreviousNode(path, lookBack = 1) {
-  return getSiblingNode(path, -lookBack);
-}
-
-function getNextNode(path) {
-  return getSiblingNode(path, 1);
+  return node.type === "TextNode" && !/\S/.test(node.chars);
 }
 
 function isPrettierIgnoreNode(node) {
   return (
-    isNodeOfSomeType(node, ["MustacheCommentStatement"]) &&
+    node?.type === "MustacheCommentStatement" &&
     typeof node.value === "string" &&
     node.value.trim() === "prettier-ignore"
   );
 }
 
 function hasPrettierIgnore(path) {
-  const { node } = path;
-  const previousPreviousNode = getPreviousNode(path, 2);
   return (
-    isPrettierIgnoreNode(node) || isPrettierIgnoreNode(previousPreviousNode)
+    isPrettierIgnoreNode(path.node) ||
+    (path.isInArray &&
+      (path.key === "children" ||
+        path.key === "body" ||
+        path.key === "parts") &&
+      isPrettierIgnoreNode(path.siblings[path.index - 2]))
```

Comment:
This part doesn't look right to me, but it's how it works before. I'll look into it.